fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java

```diff
@@ -217,6 +219,18 @@ protected OptExpression rewriteProjection(RewriteContext rewriteContext,
                 }
                 queryColumnRefToScalarMap.put(entry.getKey(), rewritten);
             }
+            for (ColumnRefOperator groupKey : queryAggregationOperator.getGroupingKeys()) {
+                ScalarOperator mapped = rewriteContext.getQueryColumnRefRewriter().rewrite(groupKey.clone());
+                ScalarOperator swapped = columnRewriter.rewriteByQueryEc(mapped);
+                ScalarOperator rewritten = rewriteScalarOperator(swapped,
```

Comment:
suggest to use Optional<T> to avoid NPE for rewriteScalarOperator

Reply:
yes, good suggestion. But I want to do it in aother pr.